src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1748,7 +1755,7 @@ private void doEncodePublish(
                 final int propertiesSize0 = propertiesSize.get();
                 final MqttPublishFW publish =
                         mqttPublishRW.wrap(writeBuffer, DataFW.FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
-                                     .typeAndFlags(0x30)
+                                     .typeAndFlags(0x30 | publishFlags)
```

Comment:
Hoist `0x03` (not `0x30`) as constant called `PUBLISH_TYPE`, then `typeAndFlags(PUBLISH_TYPE << 4 | publishFlags)`